### DISCUSSION TITLE
Add support for query conditions using `$search` for Select field with relationships

### DISCUSSION THREAD

**Discussion by takashato** [filamentphp/filament#12705] [Ideas]
Hello everyone,

I'm building an app which is using Scout / Meilisearch, so that I need accessing the `$search` value when queries the data.

When integrating it into the Select Form field, I found if we want to access the `$search` value on `$selectComponent->relationship(modifyQueryUsing)`, it is not impossible. If we want that we have to override the whole `getSearchResultsUsing` callback / closure which the `->relationship()` set.

It's because we just evaluate `$modifyQueryUsing` with only the `$query`
https://github.com/filamentphp/filament/blob/eda8452946bbae35376e01c8e80775361532d28a/packages/forms/src/Components/Select.php#L707

My idea is to pass `$search` to the evaluation so that we can access it right directly in the `$modifyQueryUsing`
<img width="596" alt="image" src="https://github.com/filamentphp/filament/assets/18548719/dfc0a285-6f7d-4c3a-a28f-2c2cfe696cb3">

Or we can update the type of `$component` from `Select` to `self` so that we can override `->applySearchConstraint`
https://github.com/filamentphp/filament/blob/eda8452946bbae35376e01c8e80775361532d28a/packages/forms/src/Components/Select.php#L697

Or better, if you think that's good I will made a small PR.

Thanks.

@danharrin

**Comment by danharrin** [comment]
Please create the PR to pass the search into the query function
